src/home/Home.tsx

```diff
@@ -1,14 +1,309 @@
-import { RouteComponentProps } from "../shared/types";
+import Loader from "../shared/components/Loader";
+import Dateline from "../shared/components/Dateline";
+import Paginator from "../shared/components/Paginator";
+import ActionLink from "../shared/components/ActionLink";
+import { LinkButton } from "../shared/components/Button";
 
-const Home: React.FC<RouteComponentProps> = (props) => {
+import "./styles/Home.scss";
+
+const taglines = [
+  /* Contributed by terrifiED */
+  "Fresh off the press",
+  "Recently proved to exist",
+  "Newly conjectured",
+  "Our latest and greatest",
+  "Thank Mr. Goose it's here",
+  "Now with 100% less fibre",
+  "Start your addiction today",
+  "Comes with free bikini",
+  "Proof of Imprint's inferiority",
+  "Bring the editors pizza day",
+  "Pissing off Imprint since 1973",
+  "The issue you've been waiting for",
+  "Just pushed to master",
+  "This issue broke prod",
+  "#1 among Russian spambots",
+  "Your statistically significant other",
+  "P(You read this) = 1",
+  "This issue > last issue",
+  "Math is fun, but we are funnier",
+  "Your assignment can wait",
+  "Now with less math and even less news",
+  "#1 cause of procrastination among mathematicians",
+  "What if you read this issue…haha jk…unless?",
+  "10/10 mathematicians recommend mathNEWS (n = 10)",
+  "The Dean loves us, apparently",
+  "Now with a chance of actual journalism",
+  // "Please read, for the jar of dirt in the office",
+  "You came here for the profQUOTES, didn't you",
+  "Made with copious amounts of coffee",
+  "Proudly published from the 7th floor of MC",
+  "I ran out of jokes to put here",
+  /* Contributed by god⚡peED */
+  "Hot and ready",
+  "Straight from the pizza oven",
+  "Putting the funny in mathematically-themed journalism",
+  /* Contributed by writer tendstofortytwo */
+  "The prime factor behind my existence",
+  "(cons mastHEAD insanity)",
+  "Freshly Imprinted for you",
+  "In a congruence class of its own",
+] as const;
+
+const IssueEntryBody: React.FC<{
+  issue: Issue;
+  dateline: string;
+}> = ({ issue, dateline }) => {
+  return (
+    <>
+      <Dateline>
+        {dateline} &bull;{" "}
+        {format(new Date(issue.publish_date || Date.now()), "d MMM y")}
+      </Dateline>
+      <h2 className="IssueEntry_title mt-1">
+        <Link to={`/issues/${issue.id}`}>{issue.title}</Link>
+      </h2>
+      <div className="IssueEntry_description mt-3">{issue.description}</div>
+    </>
+  );
+};
+
+const IssueEntry: React.FC<{ issue: Issue } & ForwardAttributes> = ({
+  issue,
+  className,
+}) => {
+  return (
+    <div className={`IssueEntry ${className || ""}`}>
+      <IssueEntryBody
+        issue={issue}
+        dateline={`Volume ${issue.volume_num} Issue ${issue.issue_code}`}
+      />
+      <div className="mt-3">
+        {/* TODO: When issue interface is changed, remove fallback */}
+        <ActionLink to={issue.pdf || ""} className="IssueEntry_cta">
```

Comment:
The `FullHero` links to the issue page, but the link for these regular entries links direct to the pdf. Is this intentional?

Reply:
Also worth noting since this renders to a `<Link>` component, the `issue.pdf` gets treated as a relative URL, and we currently return them as absolute URLs, which breaks the link.

Reply:
The link discrepancy is an oversight on my part, my bad.

Regarding `issue.pdf`, I looked into it and the API returns a relative link for the `latest` endpoint, but an absolute link on the list endpoint. Would you happen to know why?

Reply:
Seems to be [this](https://stackoverflow.com/a/52357351). The `ViewSet`s pass in a context with the Django request automatically, which lets DRF build an absolute URL. The `latest` endpoint uses a manually instantiated serializer, so it can't do this. 

There's an easy fix in the answer, basically we pass in `{ 'request': request }` to the serializer on instantiation.

Reply:
Fixed the links: ca7e995b